examples/load-reporting-service/server/lrs_server.go

```diff
@@ -0,0 +1,110 @@
+package server
+
+import (
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	gcpLoadStats "github.com/envoyproxy/go-control-plane/envoy/service/load_stats/v2"
+	"github.com/golang/protobuf/ptypes/duration"
+	"google.golang.org/grpc"
+	"log"
+	"sync"
+)
+
+// Server handling Load Stats communication
+type Server interface {
+	gcpLoadStats.LoadReportingServiceServer
+	SendResponse(cluster string, upstreamCluster []string, frequency int64)
+}
+
+func NewServer() Server {
+	return &server{lrsCache: make(map[string]map[string]NodeMetadata)}
+}
+
+type server struct {
+	// protects lrsCache
+	mu sync.RWMutex
```

Comment:
sync.RWMutex adds RLock and RUnlock for more granular reader/writer locking. However, only Lock/Unlock are ever used. Consider changing this to the simpler sync.Mutex